openassessment/xblock/ui_mixins/mfe/page_context_serializer.py

```diff
@@ -248,7 +252,16 @@ def get_activeStepName(self, instance):
         if not instance.workflow_data.has_workflow:
             return "submission"
         else:
-            return STEP_NAME_MAPPINGS[instance.workflow_data.status]
+            step_name = STEP_NAME_MAPPINGS[instance.workflow_data.status]
+            if step_name == "waiting":
+                workflow_requirements = instance.workflow_data.workflow_requirements
+                if 'staff' in workflow_requirements and workflow_requirements['staff']['required']:
+                    return "staff"
+                elif 'peer' in workflow_requirements:
+                    return "peer"
+                else:
+                    raise UnknownActiveStepException("Workflow is in waiting but no staff or peer step is required.")
+            return step_name
```

Comment:
Nit: Can remove the need to early assign step name, and use more consistent quotation marks.

```suggestion
            if instance.workflow_data.status == "waiting":
                workflow_requirements = instance.workflow_data.workflow_requirements
                if "staff" in workflow_requirements and workflow_requirements["staff"]["required"]:
                    return "staff"
                elif "peer" in workflow_requirements:
                    return "peer"
                else:
                    raise UnknownActiveStepException("Workflow is in waiting but no staff or peer step is required.")
            return STEP_NAME_MAPPINGS[instance.workflow_data.status]
```